src/main/java/com/example/online_shop_api/Service/UserService.java

```diff
@@ -325,7 +325,28 @@ public ResponseEntity<List<UserOrdersResponse>> getCurrentUserOrders(Long userId
         return ResponseEntity.ok(responseList);
     }
 
-    public ResponseEntity<String> changeOrderStatusToCancelled(Long orderId) {
+    private void checkIfOrderHasStatusBasketOrCancelled(Order order) {
+        Optional<OrderStatus> optionalBasketStatus = orderStatusRepository.findByName("BASKET");
+        Optional<OrderStatus> optionalCancelledStatus = orderStatusRepository.findByName("CANCELLED");
+        if (optionalBasketStatus.isPresent()) {
+            OrderStatus basketStatus = optionalBasketStatus.get();
+            if (order.getStatus().equals(basketStatus)) {
+                throw new OrderStatusException("Orders with status \"Basket\" cannot be cancelled!");
+            }
+        }
+        if (optionalCancelledStatus.isPresent()) {
+            OrderStatus cancelledStatus = optionalCancelledStatus.get();
+            if (order.getStatus().equals(cancelledStatus)) {
+                throw new OrderStatusException("This order is already cancelled!");
+            }
+        }
+    }
+
+    private boolean isOrderOfCurrentUser(Order order, User user) {
+        return order.getUser().getId().equals(user.getId());
+    }
+
+    public ResponseEntity<String> changeOrderStatusToCancelled(Long orderId, Authentication authentication) {
```

Comment:
Do you think that cancelOrder would be a better name for this method? I don't mind the current one too

Reply:
Ok, i will change it .